Server/routes/reactions.js

```diff
@@ -0,0 +1,326 @@
+
+const handleError = (response, err) => {
+    if (err.message === "No data returned from the query.") {
+        response.status(404)
+        response.json({
+            status: 'fail',
+            message: 'Unexpected route',
+            payload: null,
+        })
+    } else if (err.constraint === 'comments_post_id_fkey') {
+        response.status(404)
+        response.json({
+            status: 'fail',
+            message: 'Wrong route',
+            payload: null,
+        })
+    } else { 
+        console.log(err)
+        response.status(500)
+        response.json({
+            status: 'fail',
+            message: 'Sorry, Something Went Wrong (BE)',
+            payload: null,
+        })
+    }
+}
+
+const isValidId = (id) => {
+    if (!isNaN(parseInt(id)) && id.length === (parseInt(id) + '').length) {
+        return true
+    }
+    return false
+}
+
+const isUserAllowed = async (response, id, password) => {
+    try {
+        return await authenticateUser(id, password)
+    } catch (err) {
+        console.log(err)
+        if (err.message === "No data returned from the query.") {
+            return false
+        } else {
+            handleError(response, err)
+        }
+    }
+}
+
+// GET ALL REACTIONS BY POST ID
+
+router.get('/post/all/:postId', async (request, response) => {
+    const postId = request.params.postId;
+    const validId = isValidId(postId);
+
+    if (!validId) {
+        response.status(404)
+            response.json({
+                status: 'fail',     
+                message: 'Wrong route',
+                payload: null,
+            })
+    } else {
+        try {
+            const allReactionsByPostId = await getAllReactionsByPostId(postId);
+            if (allReactionsByPostId.length) {
+                response.json({
+                    status: 'success',
+                    message: `Successfully retrieved all reactions related to the post: ${postId}`,
+                    payload: allReactionsByPostId,
+                })
+            } else {
+                response.status(400)
+                response.json({
+                    status: 'fail',
+                    message: 'Wrong Route',
+                    payload: null,
+                })
+            }
+        } catch (err) {
+            handleError(response, err)
+        }
+    }
+})
+
+// GET ALL REACTIONS BY COMMENT ID
+
+router.get('/comment/all/:commentId', async (request, response) => {
+    const commentId = request.params.commentId;
+    const validId = isValidId(commentId);
+
+    if (!validId) {
+        response.status(404)
+            response.json({
+                status: 'fail',     
+                message: 'Wrong route',
+                payload: null,
+            })
+    } else {
+        try {
+            const allReactionsByCommentId = await getAllReactionsByCommentId(commentId);
+            if (allReactionsByCommentId.length) {
```

Comment:
Maybe change the conditional to something like
Array.isArray(allReactionsByPostId) instead because an empty array isn't wrong route